packages/rmb_direct_client/lib/util.ts

```diff
@@ -68,3 +70,18 @@ export function wordArrayToUint8Array(data: cryptoJs.lib.WordArray) {
   }
   return new Uint8Array(dataArray);
 }
+
+export async function getTwin(id: number, twins: Map<number, { twin: Twin; timestamp: number }>, tfclient: Client) {
+  let twin;
+  const mappedTwin = twins.get(id);
+  const isValid = moment(mappedTwin?.timestamp).isBefore(moment().subtract(10, "minutes"));
+
+  if (mappedTwin && isValid) {
+    twin = mappedTwin.twin;
+  } else {
+    twin = await tfclient.twins.get({ id });
+    twins.set(id, { twin: twin, timestamp: Math.round(Date.now() / 1000) });
+  }
+
+  return twin;
+}
```

Comment:
We can check if `mappedTwin` exists then we check if it's timestamp still valid otherwise we don't need to make 2 instances from moment js for a `false` check

Reply:
done.